### DISCUSSION TITLE
How to track git head?

### DISCUSSION THREAD

**Discussion by kugland** [renovatebot/renovate#16510] [Request Help] [answered]
I want to track git `HEAD` with renovate for one of the dependencies of a project, but I can’t get it to work.

My `renovate.json`:

```json
{
    "extends": [
        "config:base"
    ],
    "enabledManagers": ["regex"],
    "regexManagers": [{
        "fileMatch": ["^zshrc$"],
        "matchStrings": [
            "\n\\s*# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\n.*?_VERSION=(?<currentValue>.*)\n"
        ],
        "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
    }]
}
```

The relevant line in the source code:

```sh
# renovate: datasource=git-refs depName=zsh-users/zsh-history-substring-search versioning=git
ZSH_HISTORY_SUBSTR_SEARCH_VERSION=e310a75a52ed17947c11a3ecd168b4980be008b7
```

Here’s the full repo:

https://github.com/kugland/my-zshrc

**Comment by rarkins** [comment, marked as answer]
For a starter, to use `git-refs` you need to have a fully qualified URL, e.g. `https://github.com/zsh-users/zsh-history-substring-search`. The bot can't guess that when you say git you mean github.

I suggest you then try using `master` as `currentValue` and then parse the sha into `currentDigest`

**Reply by kugland** [reply]
Thanks! It worked.
